src/odemis/driver/smaract.py

```diff
@@ -4754,139 +4416,76 @@ def _doReference(self, future):
-                self.EnableEventNotification(SA_PS_STABLE_STATE_CHANGED_EVENT)
-                self.SetProperty_i32(SA_PS_AF_CHANNEL_VALIDATION_STATE_PROP, SA_SI_ENABLED)
-            self._wait_for_progress_event(SA_SI_ENABLED, timeout=600)
-            self.DisableEventNotification(SA_PS_STABLE_STATE_CHANGED_EVENT)
-
-            # Enable channels
-            for ch in self._channels.values():
-                with future._moving_lock:
-                    if future._must_stop.is_set():
-                        raise CancelledError()
-                    self.SetProperty_i32(SA_PS_CH_ENABLED_PROP, SA_SI_ENABLED, idx0=ch)
-                # Channel enabled event not triggered if it was already enabled
-                # --> poll SA_PS_CH_ENABLED_PROP attribute instead
-                while self.GetProperty_i32(SA_PS_CH_ENABLED_PROP, idx0=ch) != SA_SI_ENABLED:
-                    if future._must_stop.is_set():
-                        raise CancelledError()
-                    time.sleep(0.01)
-        except SA_SIError as ex:
-            # This occurs if a stop command interrupts referencing
-            if ex.errno == SA_SI_ERROR_CANCELLED:
-                logging.info("Validation stopped: %s", ex)
-                raise CancelledError()
-            elif future._must_stop.is_set():
-                raise CancelledError()
-            else:
-                logging.error("Validation failed: %s", ex)
-                raise
-        except CancelledError:
-            logging.debug("Validation cancelled.")
-            raise  # No fuss, pass it as-is
-        except Exception:
-            logging.exception("Validation failure.")
-            raise
-        finally:
-            if not self.IsStable():
-                raise IOError("Validation failed, device not stable.")
-            for ch, num in self._channels.items():
-                if not self.IsValid(num):
-                    raise IOError("Failed to validate channel %s" % num)
+    def _getReadableEvent(self, event):
+        """
+        Get readable event name for logging purposes.
 
-            # Start polling thread
-            self._polling_thread = util.RepeatingTimer(1, self._updatePosition, "Position polling")
-            self._polling_thread.start()
-            self._updatePosition()
+        :param error: (si.EventType or ps.EventType) The event.
+        :return: (str) The readable event name.
 
-    def _on_referenced(self, future):
-        """
-        Callback function for referencing/validation future in __init__.
         """
         try:
-            future.result()
-            self.state._set_value(model.ST_RUNNING, force_write=True)
-        except Exception as e:
-            self.state._set_value(e, force_write=True)
-            logging.exception(e)
+            ev_name = "si.EventType." + si.EventType(event.type).name + " " + "0x{:04X}".format(si.EventType(event.type).value)
```

Comment:
Why not use f-string here?
```suggestion
            ev_name = f"si.EventType.{si.EventType(event.type).name} 0x{si.EventType(event.type).value:04X}"
```